### ISSUE TITLE
FirebaseUI 4.1.1 with TwitterCore 3.0.0 causes linker failure

### ISSUE THREAD

**Issue by ghowen** [firebase/FirebaseUI-iOS#321]
With the update to FirebaseUI 4.x the dependency with TwitterCore changed from something like 2.8.x to 3.0.0. But there is an incompatibility with TwitterCore 3.0.0 that causes a linker failure:

```
  "_TWTRIdentifierForAdvertising", referenced from:
      +[TWTRCardConfiguration deviceID] in TwitterKit(TWTRCardConfiguration.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

```

Downgrading TwitterCore to 2.8.0 in the podfile with

`
pod 'TwitterCore', '~>2.8.0'
`

can be used to temporarily fix the issue.

See also
https://stackoverflow.com/questions/44188112/undefined-symbols-for-architecture-x86-64-in-xcode-after-firebase-update

iOS 10.3.3
Firebase 4.0.4
FirebaseUI 4.1.1

**Comment by morganchen12** [comment]
It seems we're currently pinning the version of TwitterKit, but not TwitterCore. The podspec for TwitterKit 2.8.1 allows all versions of TwitterCore over 2.8 including versions 3.x and above, which is probably a bug.

Generally we shouldn't be pinning versions of both TwitterKit and TwitterCore, since TwitterCore is a dependency of TwitterKit and isn't otherwise shared by FirebaseUI.

Please continue to use the workaround for now. We'll bump the version of TwitterKit to 3.0 in the next release, which should resolve this issue.

**Comment by yramanchuk** [comment]
@ghowen issue should be fixed in FirebaseUI 4.2.0

**Comment by ghowen** [comment]
Thanks! I can confirm that is is working now.
